plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/util/IcebergRegisterTableUtil.java

```diff
@@ -0,0 +1,125 @@
+        try {
+            int latestMetadataVersion = -1;
+            FileIterator fileIterator = fileSystem.listFiles(metadataDirectoryLocation);
+            while (fileIterator.hasNext()) {
+                FileEntry fileEntry = fileIterator.next();
+                if (fileEntry.path().contains(METADATA_FILE_EXTENSION)) {
+                    OptionalInt version = parseVersion(fileEntry.path());
+                    if (version.isPresent()) {
+                        int versionNumber = version.getAsInt();
+                        if (versionNumber > latestMetadataVersion) {
+                            latestMetadataVersion = versionNumber;
+                            latestMetadataLocations.clear();
+                            latestMetadataLocations.add(fileEntry.path());
+                        }
+                        else if (versionNumber == latestMetadataVersion) {
+                            latestMetadataLocations.add(fileEntry.path());
+                        }
+                    }
+                }
+            }
+            if (latestMetadataLocations.isEmpty()) {
+                throw new TrinoException(ICEBERG_INVALID_METADATA, "No versioned metadata file exists at location: " + metadataDirectoryLocation);
+            }
+            if (latestMetadataLocations.size() > 1) {
+                throw new TrinoException(ICEBERG_INVALID_METADATA, format(
+                        "More than one latest metadata file found at location: %s, latest metadata files are %s",
+                        metadataDirectoryLocation,
+                        latestMetadataLocations));
+            }
+        }
+        catch (IOException e) {
+            throw new TrinoException(ICEBERG_FILESYSTEM_ERROR, "Failed checking table's location: " + location, e);
+        }
+        return getOnlyElement(latestMetadataLocations);
+    }
+
+    public static String stripTrailingSlash(String value, String name)
+    {
+        checkNotEmpty(value, name);
+
+        String result = value;
+        for (; result.endsWith("/"); ) {
+            result = result.substring(0, result.length() - 1);
+        }
+        return result;
+    }
+
+    private static String checkNotEmpty(String value, String name)
+    {
+        if (value == null) {
+            throw new NullPointerException(name + " is null");
+        }
+        if (value.isEmpty()) {
+            throw new IllegalArgumentException(name + " is empty");
+        }
+        return value;
+    }
+
+    public static void validateMetadataFileName(Optional<String> fileName)
+    {
+        if (fileName.isEmpty()) {
+            return;
+        }
+        String metadataFileName = fileName.get();
+        checkProcedureArgument(!metadataFileName.isEmpty(), "metadata_file_name cannot be empty when provided as an argument");
+        checkProcedureArgument(metadataFileName.contains(".metadata.json"), "%s is not a valid metadata file", metadataFileName);
```

Comment:
we may omit that. let the library be picky, we don't have to be
the library will throw anyway eg on invalid/ malformed file, etc.

Reply:
Removed extra check